R/test-app.R

```diff
@@ -36,15 +37,26 @@ testApp <- function(
 
   # appDir could be the path to an .Rmd file. If so, make it point to the actual
   # directory.
-  if (is_rmd(appDir)) {
+  if (is_app(appDir)) {
+    app_filename <- NULL
+    appDir       <- appDir
+  } else if (is_rmd(appDir)) {
+    # Fallback for old behaviour
     app_filename <- basename(appDir)
     appDir       <- dirname(appDir)
     if (length(dir(appDir, pattern = "\\.Rmd$", ignore.case = TRUE)) > 1) {
       stop("For testing, only one .Rmd file is allowed per directory.")
     }
   } else {
-    app_filename <- NULL
-    appDir       <- appDir
+    rmds <- dir(appDir, pattern = "\\.Rmd$")
+    if (length(rmds) == 1) {
+      app_filename <- rmds
+    } else {
+      stop(
+        "`appDir` doesn't contain 'app.R', 'server.R', or a single '.Rmd'",
```

Comment:
Would be better to say "exactly one .Rmd". Otherwise it could be interpreted as though multiple .Rmds should work.

Reply:
Good idea